src/vs/workbench/api/node/extHostLanguageFeatures.ts

```diff
@@ -214,6 +214,33 @@ class ReferenceAdapter {
 	}
 }
 
+class ExternalReferenceAdapter {
+
+	constructor(private _provider: vscode.ExternalReferenceProvider) {
+	}
+
+	provideExternalReferences(query: modes.ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: modes.IReferenceInformation[]) => void): TPromise<modes.IReferenceInformation[]> {
+		const convertReferences = (references: vscode.ReferenceInformation[]) => {
```

Comment:
> I was using the term "rebase" loosely, but I am actually not convinced either way as far as rebasing vs merge. It is definitely a discussion worth having when we get to that point. I can see arguments for both sides.

Yeah, let's discuss this in-depth when we get there. I just wanted to throw this in before we just do it either way :)

> What do you mean? I have already done an interactive rebase on vscode/patch once to squash some commits together (to reduce cumulative diff) and it appears to have maintained the original author and commit date.

git differentiates between _commiter_ vs _author_ and _commit date_ vs _author date_. So when squashing you won't notice (especially on GitHub which shows everything linear). Squashing is great for removing useless commits like "Fix linting error", "Fix typo", etc. I love squash-merging PRs.
But when you do a bunch of commits over a period of multiple weeks (as results of multiple squash-merged PRs), and at the same time the VS Code team makes a lot of commits on upstream, if you then rebase all your history on VS Code's commit history, you loose the information that these developments/commit histories happened in parallel until they were literally "merged" at a specific point. Maybe not when looking at the timestamps, but from looking at the parent relationship between commits. Eg `git log --graph` would not be able to show you two parallel histories anymore that got merged at one or more points, it just shows you a linear history with our commits on the top, rewritten every time we need upstream changes.
Besides that, a rebase requires a force push and every developer needs to do a hard reset instead of a pull, loosing local commits if not done carefully. This is fine and I do it a lot for "own" feature branches, but on a branch that multiple people work on (and is a _target_, not a _source_ of PRs) merges should be preferred imo.